### ISSUE TITLE
Report invalid second-level command options explicitly

### ISSUE THREAD

**Issue by likebreath** [cloud-hypervisor/cloud-hypervisor#5435]
Invalid second-level command options are now simply ignored. This can cause confusions for users, particularly with options depending on non-default build features. One example is using "--platform tdx=on" without enabling the "tdx" build feature (reported by @jodh-intel). 

Explicitly reporting invalid second-level command options can improve user experience and help user identify issues earlier.

Some more context. Our command-line interface is consist of two levels. The first level commands are parsed via the `argh` crate and invalid options will be reported explicitly. For example:

```
$ ./target/debug/cloud-hypervisor --invalid                                                                                                                                                                      
Unrecognized argument: --invalid                                                                                                                                                                               
                                                                                                                                                                                                               
Run cloud-hypervisor --help for more information.
```

The second level commands are parsed via our own `option-parser` crate which will silently ignore invalid options. For example:
```
$ ./target/debug/cloud-hypervisor --platform invalid
(nothing will be reported)
```

**Comment by likebreath** [comment]
@rveerama1 I believe it is a good exercise for learning CH code base and Rust in general. Do you want to tackle this one?

**Comment by liuw** [comment]
> A separate issue: I found the help information for second-level command options is static, as it comes from the literal comments of the top-level commands. For example:
> 
> ```
>     #[argh(option, long = "platform")]
>     /// num_pci_segments=<num_pci_segments>,iommu_segments=<list_of_segments>,serial_number=<dmi_device_serial_number>,uuid=<dmi_device_uuid>,oem_strings=<list_of_strings>
>     platform: Option<String>,
> ```
> 
> As a result the "--platform tdx" option is invisible from the `--help` output no-matter if it is enabled or not (via enabling the `tdx` build feature).
> 
> @liuw Any thoughts about how we can fix this?

Hmm... there is no easy way to fix the unless we fork or hack argh.

**Comment by rafaelmsoares** [comment]
I guess that are two issues related to this:
1)  The first is related to commit 3d906985 ("main: reset tty if starting the VM fails"), which changed start_vmm() to join the vmm thread if an error happens after the vmm thread is started. The implementation put all the error-prone code that is run after the vmm is started in a closure, to be able to always join the vmm thread, regardless of any error happening. However, it missed propagating the error that might happen inside the closure back to the main function, after joining the vmm thread.

2) Some parameters, such as the --platform, are only parsed if a path to kernel or firmware is passed to the cmd line:
https://github.com/cloud-hypervisor/cloud-hypervisor/blob/eca75dcfc9babbf8479ad3963ddddaca20cccd00/src/main.rs#L536-L540C7
Thus, in the example above `./target/debug/cloud-hypervisor --platform invalid`, the platform option doesn't even get parsed.

I opened a PR to address (1).

**Comment by likebreath** [comment]
Fixed with #5453.
